### PR TITLE
Use auth libraries to authenticate in E2E tests

### PR DESCRIPTION
**Pull request by anatbal**
# Resolves [#3296](https://github.com/microsoft/AzureTRE/issues/3296)
## What is being addressed

Describe the current behavior you are modifying. Please also remember to update any impacted documentation.
- Replace the REST calls to login.microsoftonline.com with  ClientSecretCredential, UsernamePasswordCredential

### CODE REVIEW

File: e2e_tests/cloud.py [hunk 1]
```diff
@@ -5,5 +5,5 @@ def get_cloud() -> cloud.Cloud:
     return cloud.get_active_cloud()
 
 
-def get_aad_authority_url() -> str:
-    return get_cloud().endpoints.active_directory
+def get_authority_domain() -> str:
```
Comment (tamirkamara) [inline e2e_tests/cloud.py:8]: Can you verify this is the same method name we used in the other apps?

File: e2e_tests/resources/workspace.py [hunk 2]
```diff
@@ -41,24 +39,13 @@ async def get_workspace_auth_details(admin_token, workspace_id, verify) -> Tuple
...
-        try:
-            responseJson = response.json()
-        except JSONDecodeError:
-            raise Exception("Failed to parse response as JSON: {}".format(response.content))
-
-        if "access_token" not in responseJson or response.status_code != status.HTTP_200_OK:
-            raise Exception("Failed to get access_token: {}".format(response.content))
-
-        return responseJson["access_token"], scope_uri
+    return token.token, scope_uri
```
Comment (tamirkamara) [inline e2e_tests/resources/workspace.py:51]: This looks to be very similar to the block above. Is it possible to extract into a generic method like _get_token_ in the helpers class?

**Review by tamirkamara — APPROVED** [review]

File: e2e_tests/resources/workspace.py (hunk 2 above)
Comment (anatbal) [reply e2e_tests/resources/workspace.py]: Done

File: e2e_tests/cloud.py (hunk 1 above)
Comment (anatbal) [reply e2e_tests/cloud.py:8]: Yes, the exact same one as in the api, I just added the 'str' for the return value to remain consistent

Comment (anatbal) [reply e2e_tests/cloud.py]: FIixed

**Pull request merged**